default-placeholders/index.md

```diff
@@ -0,0 +1,269 @@
+
+```cpp
+namespace access {
+...
+
+enum class placeholder // deprecated
+{...};
+} // namespace access
+
+template <typename dataT,
+          int dimensions,
+          access::mode accessMode,
+          access::target accessTarget = access::target::global_buffer,
+          access::placeholder isPlaceholder = access::placeholder::false_t>
+class accessor;
+```
+
+### All accessors with a global or constant target can be placeholders
+
+SYCL 1.2.1 allows `access::placeholder::true_t`
+only when the access target is `global_buffer` or `constant_buffer`.
+We propose that the same placeholder semantics
+still apply to only these two targets,
+just that the semantics and API is available
+regardless of the `isPlaceholder` template parameter.
+
+### Accept all accessors in `handler::require`
+
+At the moment the member function `handler::require` only accepts
+placeholder accessors.
+This should stay the same,
+but since the enum class is deprecated,
+the function needs to be extended:
+
+```cpp
+class handler {
+ public:
+  ...
+
+  // Adds isPlaceholder to existing SYCL 1.2.1 function
+  template <typename dataT,
+            int dimensions,
+            access::mode accessMode,
+            access::target accessTarget,
+            access::placeholder isPlaceholder>
+  void require(accessor<dataT,
+                        dimensions,
+                        accessMode,
+                        accessTarget,
+                        isPlaceholder>
+                acc);
+};
+```
+
+### Deprecate `is_placeholder`
+
+The function `accessor::is_placeholder` doesn't make sense anymore,
+we propose deprecating it.
+
+### New constructors
+
+We propose adding new constructors to the accessors class
+to allow placeholder construction.
+
+1. Default constructor - not part of SYCL 1.2.1,
+   but there is a [Codeplay proposal](https://github.com/codeplaysoftware/standards-proposals/pull/89)
+   for making that happen.
+1. 0-dim accessor constructor from a buffer -
+   normally an accessor constructor requires a buffer and a handler,
+   we propose making the handler optional.
+   This is the same constructor currently allowed for host buffers.
+1. Constructor from a buffer -
+   normally an accessor constructor requires a buffer and a handler,
+   we propose making the handler optional.
+   This is the same constructor currently allowed for host buffers.
+
+```cpp
+template <typename dataT,
+          int dimensions,
+          access::mode accessMode,
+          access::target accessTarget = access::target::global_buffer,
+          access::placeholder isPlaceholder = access::placeholder::false_t>
+class accessor {
+ public:
+  ...
+
+  // 1
+  // Only available when ((accessTarget == access::target::global_buffer) ||
+  //                      (accessTarget == access::target::constant_buffer))
+  accessor() noexcept;
+  
+  // 2
+  // Only available when: ((accessTarget == access::target::global_buffer) ||
```

Comment:
I'm not sure we should include `access::target::host_buffer` in the new constructors (at least for now), there isn't a mechanism atm for requiring a host `accessor` since it currently blocks dependencies on construction.

Reply:
You're right, I combined them because the signatures are the same, but they're actually different constructors